addons/stock_account/models/product.py

```diff
@@ -8,17 +8,48 @@
 from collections import defaultdict
 from datetime import datetime
 
+from odoo.addons.stock_account import const
+
 
 class ProductTemplate(models.Model):
     _inherit = ['product.template']
 
-    cost_method = fields.Selection(related="categ_id.property_cost_method", readonly=True)
-    valuation = fields.Selection(related="categ_id.property_valuation", readonly=True)
+    cost_method = fields.Selection(
+        string="Cost Method",
+        selection=const.COST_METHOD,
+        compute='_compute_cost_method',
+    )
+    valuation = fields.Selection(
+        string="Valuation",
+        selection=const.VALUATION,
+        compute='_compute_valuation',
+    )
     lot_valuated = fields.Boolean(
         "Valuation by Lot/Serial number",
         help="If checked, the valuation will be specific by Lot/Serial number.",
     )
 
+    @api.depends("categ_id.property_cost_method")
```

Comment:
```suggestion
    @api.depends_context('company')
    @api.depends("categ_id.property_cost_method")
```

Those two fields values won't be the same according to the company of the environment and must be correctly labeled as such (it is done automatically for company_dependent fields, but has to be done manually for standard computed fields).